src/main/php/PHPMD/TextUI/Command.php

```diff
@@ -183,12 +183,12 @@ public function run(CommandLineOptions $opts, RuleSetFactory $ruleSetFactory): E
      */
     private function getVersion(): string
     {
-        $build = __DIR__ . '/../../../../../build.properties';
+        $build = __DIR__ . '/../../../../../CHANGELOG';
 
         $version = '@package_version@';
         if (file_exists($build)) {
-            $data = @parse_ini_file($build);
-            $version = $data['project.version'] ?? $version;
+            $changelog = file_get_contents($build) ?: '';
```

Comment:
Ah I didn't noticed that it may also happen that we read the file at runtime.

CHANGELOG can become big so maybe worth to limit the number of bytes that we read:

1 kB should be enough I think:
```suggestion
            $changelog = file_get_contents($build, false, null, 0, 1024) ?: '';
```